srcpkgs/xbps-triggers/files/dkms

```diff
@@ -21,19 +21,30 @@ UPDATE="$5"
 export PATH="usr/bin:usr/sbin:/usr/sbin:/usr/bin:/sbin:/bin"
 
 remove_modules() {
-	local _modver _kver
+	local _modver _kver _line
 
 	# Remove the specified modules from all kernels.
 	set -- ${dkms_modules}
 	while [ $# -gt 0 ]; do
-		$DKMS status -m "$1" | while read line; do
-			if $(echo "$line" | egrep -vq '(added|built|installed)'); then
+		$DKMS status -m "$1" | while IFS=' ' read _modver _kver _line; do
+			if echo "$_modver $_kver $_line" | grep -Evq '(added|built|installed)'; then
 				shift 2; continue
 			fi
-			_modver=$(echo "$line"|sed "s/$1,[[:blank:]]\([^,]*\)[,:].*/\1/;t;d")
-			_kver=$(echo "$line"|awk '{print $3}'|sed "s/\(.*\),$/\1/")
-			echo -n "Removing DKMS module '${1}-${_modver}' for kernel-${_kver}... "
-			$DKMS remove -m "$1" -v "${_modver}" -k "${_kver}" >/dev/null 2>&1
+			_modver=${_modver##*/}
+			_kver=${_kver%,}
+			if [ "$_kver" = added ]; then
+				# The module wasn't built successfully for any kernel version, so it's status is 'added'
+				# The output must've been like this -
+				# 	module-lipsum/x.y.z: added
+				# Which caused _kver to be set to "added"
+				_modver=${_modver%:};
+				echo -n "Cleaning up unbuilt DKMS module '${1}-${_modver}'... "
+				$DKMS remove -m "$1" -v "${_modver}" >/dev/null 2>&1
+			else
+				_modver=${_modver%,};
+				echo -n "Removing DKMS module '${1}-${_modver}' for kernel-${_kver}... "
+				$DKMS remove -m "$1" -v "${_modver}" -k "${_kver}" >/dev/null 2>&1
+			fi
```

Comment:
```suggestion
			if [ "${_line}" = added ]; then
				# The module wasn't built successfully for any kernel version
				echo -n "Cleaning up DKMS module '${_modname}-${_modver}'... "
				$DKMS remove -m "${_modname}" -v "${_modver}" >/dev/null 2>&1
			else
				echo -n "Removing DKMS module '${_modname}-${_modver}' for kernel-${_kver}... "
				$DKMS remove -m "${_modname}" -v "${_modver}" -k "${_kver}" >/dev/null 2>&1
			fi
```

Reply:
Okay.

But what if `$1` and `$_modname` are different. Should we check for that? How should we handle that?

Reply:
The trigger already fails to verify that the module name reported by DKMS, so let's not worry about it. If you really think this could be an issue, verify that `"$1" = "${_modname}"` or skip the iteration.

My other suggestion had two bugs: 1. the shift shouldn't be applied because it is already done outside the loop (this is also a bug in the existing `shift 2; continue` line); 2. I'm not setting the kernel version properly. I think the proper solution is something more like

```sh
remove_modules() {
	local _modname _modver _kver _line _arch _status

	# Remove the specified modules from all kernels.
	set -- ${dkms_modules}
	while [ $# -gt 0 ]; do
		$DKMS status -m "$1" | while IFS="/, " read _modname _modver _line; do
			[ "$1" = "${_modname}" ] || continue

			_modver="${_modver%:}"

			if [ "${_line}" = added ]; then
				# The module wasn't built successfully for any kernel version
				echo -n "Cleaning up DKMS module '${_modname}-${_modver}'... "
				$DKMS remove -m "${_modname}" -v "${_modver}" >/dev/null 2>&1
				continue
			fi

			read _kver _arch _status <<-EOF
			${_line}
			EOF

			case "${_status}" in
				added|built|installed) ;;
				*) break ;;
			esac

			echo -n "Removing DKMS module '${_modname}-${_modver}' for kernel-${_kver}... "
			$DKMS remove -m "${_modname}" -v "${_modver}" -k "${_kver}" >/dev/null 2>&1
                done
                shift 2
        done
}


```

Reply:
Your above example doesn't print `done.` or `FAILED!` after the `... ` to indicated whether the action succeeded or failed.